### PR TITLE
content(artwork) Created an image for the outfit Puny.

### PR DESCRIPTION
**Pull request by xobes**
**Content (Artwork)**

## Summary
Puny deserved a picture.  When @tekker2234 provided my a pilot file to help debug PR #10743, they exposed to me that an outfit was missing an image, so I took it upon myself to find a suitable image based on the outfit's lore.

## Screenshots
| Before | After |
|--------|--------|
| image | image | 

Oh no.  I left evidence that I sold Puny to the outfitter in those images... 😢; I went back for her.

## Artwork Checklist
(If any artwork was added or changed by this PR, the following must be provided.)
 - [x] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [x] I created a PR to the [endless-sky-assets repo](https://github.com/endless-sky/endless-sky-assets) with the necessary image, blend, and texture assets: https://github.com/endless-sky/endless-sky-assets/pull/189
 - [x] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: https://github.com/endless-sky/endless-sky-high-dpi/pull/432
Source material: https://github.com/endless-sky/endless-sky-assets/pull/189

<details>
  <summary>Stuff I tried...</summary>
I have added several collars to the outfits, and assigned one to "Puny".

<section>
<description>
| collar1.png | collar2.png | collar3.png | collar4.png | collar5.png | collar6.png |
|--------|--------|--------|--------|--------|--------|
| collar1 | collar2 | collar3 | collar4 | collar5 | collar6 |

| collar1b.png | collar2b.png | collar3b.png | collar4b.png | collar5b.png | collar6b.png |
|--------|--------|--------|--------|--------|--------|
| collar1b | collar2b | collar3b | collar4b | collar5b | collar6b |

</details>
Long live puny image.

### CODE REVIEW

**Comment by tekker2234** [comment]
> Oh no.  I left evidence that I sold Puny to the outfitter in those images... 😢 

😦 You monster! I'll have your head for this! (/jk)

**Comment by Hecter94** [comment]
Mug has a sprite now
https://github.com/endless-sky/endless-sky/pull/9498

**Comment by Saugia** [comment]
This has come up many times in the past, alongside portraits. The TLDR is that we want to avoid adding realistic organic images to the game as they conflict with its arcade-like style in nature. Yes, there is the one with the FW mechanic, but we kind of have left that one in as an exception. Portraits used to be in the game, but were also deemed incoherent with the rest of things.

In order to work around an outfit image for Puny against all the other outfits, it would have to be modeled similarly, and modeling organics to match the in-game sprites leads is difficult in itself. In order to maintain consistency in visuals, we've declined this and other similar cases in the past.

**Comment by Saugia** [comment]
As an alternative, it was brought up that something more appropriate would be a modelled collar or bone would be an appropriate sprite for the outfit. That hasn't been attempted yet to my knowledge.

**Comment by xobes** [comment]
I'm bummed.  I did quite a bit to that image to make it less photographic and also saw a whole folder of characters including humans and an iguana and a red squirrel and a tortoise.

None of those other animals really fit the bill, but this matches the text perfectly. 

😭

**Comment by Saugia** [comment]
The folder you speak of is the original portraits that I mentioned above for each alien race and humans. We felt better not to have these directly represented due to the above, preferring to leave things to the player's imagination.

**Comment by xobes** [comment]
@Saugia can we re-open these, please?  I have addressed all of your concerns.

**Comment by Saugia** [comment]
I took a peak, and nice, you did get it modeled in blender! I'm thinking on second thought, this *could* be fine as a one-of exception; with the image, it was more a concern that it was slapping a realistic photo into a category of modelled outfits, and I think the sprite now has that sort of blended feel compared to something pulled out of, say a landscape image (in which we avoid organics there too). At the very least, we know this is a dog, and it's not some take on an alien better suited for the imagination.

Modelling biologicals is very iffy as many blender artists have different methods. While that's something that can be upkept with more simpler industrial shapes and lighting, trying to match organics is a pretty big rabbit hole and will look odd in the end, and we want to keep in-game graphics as consistent as possible.

Puny, even as an outfit with an "unknown" image, is more supposed to be that one off silly mission, so I think we could make this work, but I want to avoid future cases and repeat examples. I don't think we should have an outfitter full of a bunch of different animal images, for example, or several storylines that lead to something similar (as might be the case of Saga's Saga linked above), when outfits are mostly meant to be parts of a ship or tools, and not common companions.

I'd like to wait for other developers to chime in as well before considering anything, but I'll happily reopen these and wait for more input other than my own.

**Comment by xobes** [comment]
Sounds good to me.  Yeah, we have the file now, so if we want to change things it's certainly possible.

I didn't introduce the concept of Puny, I just felt they were deserving of some better representation.  And yeah, it's a dog, not some alien species that is better left to the imagination.

I also had the opportunity to learn some Blender over the past several days, so thanks for the challenge.

**Comment by Hecter94** [comment]
> Puny, even as an outfit with an "unknown" image, is more supposed to be that one off silly mission, so I think we could make this work, but I want to avoid future cases and repeat examples. I don't think we should have an outfitter full of a bunch of different animal images, for example, or several storylines that lead to something similar (as might be the case of Saga's Saga linked above), when the outfitter is a place for parts of a ship, tools, or materials, and not common companions.

Perhaps the best option would be to have a generic "companion" sprite that can be shared amongst all the unique companion outfits. This would avoid the need to have unique images for each of them and the issue of having "undefined" sprites for them, which can look unfinished and shoddy.

**Comment by Saugia** [comment]
>Perhaps the best option would be to have a generic "companion" sprite that can be shared amongst all the unique companion outfits. This would avoid the need to have unique images for each of them and the issue of having "undefined" sprites for them, which can look unfinished and shoddy.

This is why I kind of prefer the collar model suggestion over a model of specifically the companion, since it's more fitting in an outfitter setting and in such a case I think having multiple "collars" or "companion jewelry" would be acceptable vs. full bodies, but I'll see what others have to say.

**Comment by bene-dictator** [comment]
I appreciate the effort, but I also feel as if it'd be better to leave Puny up to the player's imagination. A collar would be appropriate for that.

**Comment by xobes** [comment]
Source material: https://github.com/endless-sky/endless-sky-assets/pull/189

I have added several collars to the outfits, and assigned one to "Puny".

| collar1.png | collar2.png | collar3.png | collar4.png | collar5.png | collar6.png |
|--------|--------|--------|--------|--------|--------|
| collar1 | collar2 | collar3 | collar4 | collar5 | collar6 |

| collar1b.png | collar2b.png | collar3b.png | collar4b.png | collar5b.png | collar6b.png |
|--------|--------|--------|--------|--------|--------|
| collar1b | collar2b | collar3b | collar4b | collar5b | collar6b |

**Comment by Hecter94** [comment]
Is this really your first time using blender? This and the original Pookie blend are some very solid work.

**Comment by xobes** [comment]
Yeah.  Thanks!

**Comment by nylki** [comment]
Solid model, i do like the star pendant as a little detail.

**Comment by Saugia** [comment]
Thank you for being patient with us by the way. I know this PR has been a bit of work and a rollercoaster, and I appreciate your flexibility.

**Comment by xobes** [comment]
> Thank you for being patient with us by the way. I know this PR has been a bit of work and a rollercoaster, and I appreciate your flexibility to work with us.

I've learned a bunch of new skills with this effort.  Happy to tweak as needed.

I envisioned collar3 for Puny.

I can remove the rest, and fiddle with the lighting.  If desired we can add a .xcf with some texturing like for other models, I think you're picking up on that more than the blender lighting...  I struggled to get the star pendant to have a good sheen and not appear dull, and so the lighting was tweaked for that purpose foremost. 

I think that this whole thought process of we don't want more companion outfits is a bit of a downer.  Why poo-poo on that so much?  The more variety and fun 'collect all X trinkets' stuff, the better.

**Comment by Saugia** [comment]
The outfitter is just that - an outfitter. It's meant for outfitting your ship/designated to parts of a ship. It does have the side use for selling minerals as well, but the majority of it is for building a ship. Pookie, when it was added, was the sort of easter-egg one of, with the outfit being this funny side thing of you having a dog in your ship. As with other easter eggs, it was meant to be a one-of case, though I'm saying there may be other cases we can give an exception, but the more exceptions we give, the less unique Pookie is. It's similar to things like the Shield Beetle Penant, and even ships like the Emerald Sword. The more unique things we add, the less unique they become. One of the aspects of ES we lean towards is staying away from rewards, and letting the stories do the pulling of narrative specials.

TLDR: The outfitter is meant for ship parts, not companions, with Pookie being this fun little exception. We don't want to standardize there being a bunch of funny little exceptions. It's not so much "poo-pooing" on it (the concept in itself is fun and entertaining), but making sure that we don't change the outfitter into something that accommodates much more than what it's intended use is for.

**Comment by Saugia** [comment]
It'd also be fair to ask @Anarchist2's thoughts as well since they added Pookie.

**Comment by xobes** [comment]
I added texture to the collar material and took the lighting/camera arrangement from `afterburner large.blend` and this is what I got, with no adjustments, this is the raw render:



| 1x | 2x |
|--------|--------|
| collar3 | collar3@2x |

**Comment by Saugia** [comment]
Lighting looks much better.

**Comment by Hecter94** [comment]
Could these sharp edges be slightly rounded or beveled? They look very artificial to me.
image

**Comment by xobes** [comment]
&

**Comment by xobes** [comment]
It's pretty darn big, now that I look at it:
image

**Comment by tekker2234** [comment]
I like the 110, I feel like the 75 is harder to make out. It being a smaller dog I wouldn't really want the collar to be too large either. (especially if another dog or other pet is added which is larger.)

**Comment by Arachi-Lover** [comment]
110 looks the best to me as well.

**Comment by xobes** [comment]
Re-rendered at 110 (and 220).  All done. 🤞

**Comment by Anarchist2** [comment]
The lighting still isn't right. It's hard to tell when it's by itself in the Uniques category, but putting it alongside other outfits, it's clear that the shadows are too bright.

image

The spikes on the collar aren't reflecting light as they should. Even on the parts of the collar where they should be entirely lit up, they're nearly black.

The material of the collar itself is too matte. Even the roughest surfaces will have some reflective highlights, and leather is a relatively reflective material itself.

**Comment by Anarchist2** [comment]
If you use a HSV colour picker on the shadows of the fuel pod, you'll see that the half of the fuel pod that is in shadow only has a lightness value of 2%. Compare this to the pink shadowed parts of the collar, which have a brightness of 23%. Make sure your world in the Blender file isn't producing any light.

Just because something is "shiny" doesn't mean that it's brighter. A mirror is a very shiny object, but unless it's being lit from a very specific angle relative to the viewer (or it's being lit from every angle), it won't look lit up. In contrast, if you point a flashlight at a piece of canvas, no matter where you stand you can perceive that it's being lit up. The metal spikes on the collar work the same way. If you don't give them enough roughness, they won't be able to reflect enough light.

If it is meant to be cloth, then why is it perfectly round? Unlike leather, fabric will crease, fold and collapse if you attempt to bend it into a circular shape. A cloth collar wouldn't be able to support metal spikes either, unless it has some sort of backing material. It's also a lot thicker than would be normal if it was made of fabric.

**Comment by xobes** [comment]
Throwing in the towel. 

Game vs realism.

**Comment by xobes** [comment]
> I'm running with a poodle from some ddg query, because the default thumbnail is simply unacceptable. 
a) what is "ddg query"

>As in- this PR's suggestion doesn't quite fit ER to ~2%, the current image doesn't quite fit ER to ~2000%.
b) I don't understand "As in- "
c) "this PR's suggestion" could mean any of about 10 things at this point.
d) What is "ER"?

Also, I closed this because I gave up on making people happy with an appropriate level of realism for a game and apparently my guess at what the right lighting should be (even going so far as to pull the exact lighting sources from another object) was in sufficient.  Unlike code reviews where reviewers leave suggested improvements, it seems with artwork we'd rather just poo-poo on things and provide more problems without suggesting solutions.

Take anything I've made and try to move it along, I have invested too much of myself and my time to withstand this crowd any longer.  I also have code reviews out there that just lay stagnant and all my beginner energy has dried up and go away.  This project is a downer.
